### ISSUE TITLE
Mysql Table debugkit LIMIT use in subquery not supported

### ISSUE THREAD

**Issue by markusramsak** [cakephp/debug_kit#730]
CakePHP Version: 3.8.8
DebugKit Version: 3.21.0

Ich have Database Lock errors with SQLLite und PHP 7.4.1 so I tried to switch to the MySQL-Database 5.7 but there are is no LIMIT clause in a subquery allowed which is used in RequestsTable::gc()

Solution: Use ->extract('id')->toArray() after the limit clause and the problem is fixed

Error: SQLSTATE[42000]: Syntax error or access violation: 1235 This version of MySQL doesn't yet support 'LIMIT & IN/ALL/ANY/SOME subquery'

**Comment by markstory** [comment]
Sounds like you should use SQLite for the DebugKit's storage instead :smile:

**Comment by markusramsak** [comment]
> What do you mean by you were running into database locking errors with SQLite? Are you running DebugKit in a multi-user environment?

I use CakePHP with DebugKit since 2011. I always used MAMP and SQLLite without any issues. But with the newest MAMP Version 5.6 (https://www.mamp.info/en/release-notes/mac/) which uses PHP 7.4.1 I got the error: "SQLSTATE[HY000]: General error: 5 database is locked". I don't know why so I tried to switch to MySQL. But than I got the posted issue ...
